src/bloom/hierarchical.rs

```diff
@@ -0,0 +1,610 @@
+use std::{borrow::Cow, collections::BTreeSet};
+
+use super::*;
+
+#[derive(Hash, Clone, Copy, Default, Debug, PartialEq, Eq, PartialOrd, Ord)]
+/// Represents id of inner in collection
+pub struct InnerId(u64);
+
+#[derive(Hash, Clone, Copy, Default, Debug, PartialEq, Eq, PartialOrd, Ord)]
+/// Represents id of child in collection
+pub struct ChildId(u64);
+
+impl ChildId {
+    fn get_inc(&mut self) -> Self {
+        let ret = *self;
+        self.0 += 1;
+        ret
+    }
+}
+
+impl InnerId {
+    fn get_inc(&mut self) -> Self {
+        let ret = *self;
+        self.0 += 1;
+        ret
+    }
+}
+
+/// Container for types which is support bloom filtering
+pub struct HierarchicalBloom<Child> {
+    inner: BTreeMap<InnerId, HierarchicalBloomInner>,
+    children: BTreeMap<ChildId, Leaf<Child>>,
```

Comment:
Is it possible to declare `children` as a vector, so that `ChildId` be an index inside that vector?

Reply:
Yeah, I've done it